cluster-operator/src/main/java/io/strimzi/operator/cluster/operator/assembly/KafkaMetadataStateManager.java

```diff
@@ -172,7 +172,7 @@ public KafkaMetadataConfigurationState getMetadataConfigurationState() {
      */
     public void checkMigrationInProgress(Reconciliation reconciliation, Secret clusterCaCertSecret, Secret coKeySecret, String controllerPodName) {
         if (this.kafkaAgentClient == null) {
-            this.kafkaAgentClient = new KafkaAgentClient(reconciliation, reconciliation.name(), reconciliation.namespace(), clusterCaCertSecret, coKeySecret);
+            this.kafkaAgentClient = initKafkaAgentClient(reconciliation, clusterCaCertSecret, coKeySecret);
```

Comment:
This seems wrong - although it may not be an issue of this PR, but it inherits the logic already - however:
* This seems to be the only place where the client is used. So why do we have it as a field in the class and not just a local variable or a parameter in the method itself?
* If it needs to be part of the whole class, why not pass the instance in a constructor?

Reply:
That's a good point. My intention was to avoid creating the `KafkaAgentClient` each time there was the need for check the migration status and a similar approach is used in the `KafkaRoller` but now I realized that, because the check for migration status on the Kafka Agent happens on each reconciliation (and not more times in the same reconcile), a new state manager is created each time, so there is no previous instance of the agent client.
It's enough to be just a local variable here. I will make this change.
The init method was instead useful to Shubham for mocking.

Reply:
Yeah, I'm not strictly against the init method. I'm not strictly against having it as a class field either even if it is used just once. But the way it is it doesn't work together I think.
* If it is a class field, it should be passed in constructor and be final. That way it can be easily passed in tests as a mock as well.
* If it is a local field, I guess I can live with the init method. But I wonder if passing it into the method itself would be nicer and make it easier to test. But I'm not 100% sure whether this is called form in the production code, so maybe the init method is the best solution here.

Reply:
There is no actual place in the operator where a `KafkaAgentClient` instance is created and provider down the line of different components. It's only used in the `KafkaRoller` which instatiates it internally, so I think that it's the way to go in this case as well, instead of creating outside and passing it down. Its usage is limited.

Reply:
I made the client local.